src/Quote/StrictQuoteLine.php

```diff
@@ -88,6 +88,9 @@ public function quoteValue($row)
                 // Elements of the array - the numbers are formatted as usual, and the dates, dates-with-time, and lines are in
                 // single quotation marks with the same screening rules as above.
                 // as in the TabSeparated format, and then the resulting string is output in InsertRow in double quotes.
+                $value = array_map(function($v) {
```

Comment:
Also, if `'"That work"'` in `['2018-04-01', '"That work"']` works, wouldn't it be better to find out why (what is the logic behind quotes there), extract the logic into single private method and then leverage the method here instead of introducing new `str_replace`? 

Maybe I'm wrong but that's the idea.

Reply:
I just want to fix bug and I dont have enough time to do the best pull request

Reply:
I'm not the owner of this project but with such approach the code would be soon very hard to maintain and fixing other bugs would require much more effort. 

If everyone adds his fix here and other fix there without following basic programming principles (DRY in this case – why adding new quote-handling logic if there's probably such logic already and only waiting to be leveraged. Especially when we're talking about array containing primitive types that are handler properly already.) it would terminatelly lead to unmaintainable code.

IMO the investigation and eventual implementation of what I proposed would take 15m.

Reply:
Nice, so I propose to create something like `function parseString()` that would be then applied to both string values and string values in array in your map function.

Reply:
Basically extracting your `str_replace()` to it should work well for now. The function would seem pretty simple but may grow more complex in the future.

I have already encountered similar problem in this library https://github.com/smi2/phpClickHouse/commit/d8941fc9bc5cf7f1f6a544348f0302f290801a27#diff-8a74139f53d8e7ff60d9e40f45200e02R130

Reply:
Right now this lib contain serious error, that PR fix bug, may be this is not best PR, but without that fix, lib cant works correct, if you want you can improve PR or do new PR and fix bug

Reply:
Yes, that's correct and you're pretty close to fixing it. You already have a working test.

What is so hard on creating a function, that properly formats strings, and then  use it for array and string type values?

I'm pretty sure no one is eager to fix spaghetti code 🍝 so introducing more spaghetti code with excuse that someone will fix it, maybe, is not very good approach in my opinion.

Reply:
How fix this error:
`Function unlink() should not be referenced via a fully qualified name, but via a use statement.`